src/main/java/com/teamacronymcoders/base/materialsystem/MaterialSystem.java

```diff
@@ -105,6 +105,18 @@ public static void registerMaterialPart(MaterialPart materialPart) {
     public static MaterialPart getMaterialPart(String name) {
         return Optional.ofNullable(materialPartMap.get(name.toLowerCase(Locale.US))).orElse(MISSING_MATERIAL_PART);
     }
+    
+    public static void removeMaterialPart(String name) {
+        if (materialPartMap.containsKey(name)) {
+            materialPartMap.remove(name);
+        }
+    }
+    
+    public static void removeMaterialPart(MaterialPart materialPart) {
+        if (materialPartMap.containsKey(materialPart.getUnlocalizedName())) {
```

Comment:
I don't believe you need to check it has something before you try and remove it, do you?